ios/Classes/TiNfcVTagTechnology.m

```diff
@@ -9,4 +9,701 @@
+  unsigned int dataValue[message.count];
+  for (int i = 0; i < message.count; i++) {
+    dataValue[i] = [message[i] unsignedIntValue];
+  }
+  NSData *data = [[NSData alloc] initWithBytes:dataValue length:message.count];
+
+  [[self.tagProxy asNFCISO15693Tag] challengeWithRequestFlags:requestFlags
+                                        cryptoSuiteIdentifier:[cryptoSuiteIdentifier integerValue]
+                                                      message:data
+                                            completionHandler:^(NSError *_Nullable error) {
+                                              if (![self _hasListeners:@"didChallengeWithRequestFlags"]) {
+                                                return;
+                                              }
+                                              [self fireEvent:@"didChallengeWithRequestFlags"
+                                                   withObject:@{
+                                                     @"errorCode" : error != nil ? NUMINTEGER([error code]) : [NSNull null],
+                                                     @"errorDescription" : error != nil ? [error localizedDescription] : [NSNull null],
+                                                     @"errorDomain" : error != nil ? [error domain] : [NSNull null]
+                                                   }];
+                                            }];
+}
+
+- (void)extendedFastReadMultipleBlocksWithRequestFlag:(id)args
+{
+  NSNumber *requestFlagsValue = [[args firstObject] valueForKey:@"requestFlags"];
+  uint8_t requestFlags = [requestFlagsValue unsignedCharValue];
+
+  [[self.tagProxy asNFCISO15693Tag] extendedFastReadMultipleBlocksWithRequestFlag:requestFlags
+                                                                       blockRange:NSMakeRange(0, 8)
+                                                                completionHandler:^(NSArray<NSData *> *_Nonnull dataBlocks, NSError *_Nullable error) {
+                                                                  if (![self _hasListeners:@"didExtendedFastReadMultipleBlocksWithRequestFlag"]) {
+                                                                    return;
+                                                                  }
+                                                                  [self fireEvent:@"didExtendedFastReadMultipleBlocksWithRequestFlag"
+                                                                       withObject:@{
+                                                                         @"errorCode" : error != nil ? NUMINTEGER([error code]) : [NSNull null],
+                                                                         @"errorDescription" : error != nil ? [error localizedDescription] : [NSNull null],
+                                                                         @"errorDomain" : error != nil ? [error domain] : [NSNull null]
```

Comment:
you should send 1 more payload dataBlocks and  with event.

Reply:
Done